libraries/core_libs/network/rpc/eth/Eth.cpp

```diff
@@ -159,9 +160,14 @@ class EthImpl : public Eth, EthParams {
     return toJS(ret.code_retval);
   }
 
-  string eth_estimateGas(const Json::Value& _json) override {
+  string eth_estimateGas(const Json::Value& _json, const string& blockNumber) override {
     auto t = toTransactionSkeleton(_json);
-    auto blk_n = final_chain->last_block_number();
+    EthBlockNumber blk_n;
+    if (!blockNumber.empty()) {
```

Comment:
If it was done because of the tests I think it is better to specify "latest" there and don't have this if